src/util/pg.py

```diff
@@ -1339,26 +1339,56 @@ def drop_depending_views(cr, table, column):
 
 
 def create_m2m(cr, m2m, fk1, fk2, col1=None, col2=None):
+    """
+    Ensure a m2m table exists or is created.
+
+    This function creates the table associated to a m2m field.
+    If the table already exists, :func:`~odoo.upgrade.util.pg.fixup_m2m` is run on it.
+    The table name can be generated automatically, applying the same logic of the ORM.
+    In order to do so, use the value "auto" for the `m2m` parameter.
+
+    :param str m2m: table name to create,
+                    if :const:`~odoo.upgrade.util.misc.AUTOMATIC` it is auto-generated
+    :param str fk1: first foreign key table name
+    :param str fk2: second foreign key table name
+    :param str col1: column referencing `fk1`, defaults to `"{fk1}_id"`
+    :param str col2: column referencing `fk2`, defaults to `"{fk2}_id"`
+    :return: the name of the table just created/fixed-up
+    :rtype: str
+    """
     if col1 is None:
         col1 = "%s_id" % fk1
     if col2 is None:
         col2 = "%s_id" % fk2
 
+    if m2m is AUTOMATIC:
+        m2m = "{}_{}_rel".format(*sorted([fk1, fk2]))
+
     if table_exists(cr, m2m):
         fixup_m2m(cr, m2m, fk1, fk2, col1, col2)
-        return
+        return m2m
 
     cr.execute(
-        """
-        CREATE TABLE {m2m}(
-            {col1} integer NOT NULL REFERENCES {fk1}(id) ON DELETE CASCADE,
-            {col2} integer NOT NULL REFERENCES {fk2}(id) ON DELETE CASCADE,
-            PRIMARY KEY ({col1}, {col2})
-        );
-        CREATE INDEX ON {m2m}({col2}, {col1});
-    """.format(**locals())
+        format_query(
```

Comment:
Note that `query = """...`
then `cr.execute(format_query(query...))` would keep the history better than indenting the whole thing.